src/bento/components/bento-mega-menu/1.0/component/Shim.tsx

```diff
@@ -0,0 +1,95 @@
+import {useEffect} from '#preact';
+import {memo} from '#preact/compat/external';
+import type {
+  EventHandler,
+  FunctionalComponent,
+  HTMLAttributes,
+  RefObject,
+} from '#preact/types';
+
+export type ShimProps = {
+  elementRef: RefObject<HTMLElement>;
+};
+/**
+ * This Shim takes a DOM element,
+ * and manually applies the props to it,
+ * using setAttribute and addEventListener.
+ */
+const Shim: FunctionalComponent<ShimProps> = memo(
+  ({elementRef, ...props}: ShimProps) => {
```

Comment:
Is there a difference between using this `elementRef` prop vs. creating a `ForwardRef` component with a ref?

Reply:
The `elementRef` here is like an "input", unlike the forwardRef `ref` which is like an "output". So it needs to be a separate property.